library-best-practices/glossary.md

```diff
@@ -0,0 +1,84 @@
+Java Dependency Glossary
+------------------------
+
+### Types of conflicts and compatibility
+
+- **Linkage conflict**: The signature of a non-private method or class in a
+  dependency has changed in an incompatible way between the version supplied at
+  compile time and the version invoked at runtime. For example, a public method
+  may be removed from a class or a class may be made final. Linkage conflicts
+  detected at runtime manifest as ReflectiveOperationException,
+  `NoClassDefFoundError`, `NoSuchFieldException`, `MethodNotFoundException`,
+  `LinkageError`, or other related exceptions.
+  - Or, another perspective: In cases where binary compatibility and source
+    compatibility are the same, a linkage conflict is when compilation would
+    fail if the libraries in the classpath were all built together from their
+    originating source code, or when reflection would fail.
+  - Opposite: **Linkage-compatible**.
+  - Sub-type: **Static linkage conflict**: A linkage conflict caused by a direct
+    code reference (non-reflective).
+  - Sub-type: **Reflective linkage conflict**: A linkage conflict caused by a
+    reflective reference.
+- **Behavior conflict**: The class's implementation has changed in a way that
+  can break clients at runtime although all signatures remain compatible. For
+  example, if a method that has been returning mutable lists begins returning
+  immutable lists without updating its signature, dependents that mutate the
+  list will fail, possibly far away from the original call site. By contrast, a
+  change in return type from List to ImmutableList would be a linkage conflict.
+  - Opposite: **Behavior-compatible**.
+- **Version disagreement**: This is a class of failure reported by build systems
```

Comment:
Can we add some blank lines in between paragraphs for clarity or will that mess up the formatting?

Reply:
I think the formatting will be unaffected because it is an unordered list.

Another option is for you to get used to list items sitting next to each other :-)